controllers/v1/solutions.js

```diff
@@ -842,13 +842,13 @@ module.exports = class Solutions extends Abstract {
   async verifyLink(req) {
     return new Promise(async (resolve, reject) => {
       try {
-        
         let solutionData = await solutionsHelper.verifyLink(
           req.params._id,
           req.body,
           req.userDetails.userId,
           req.userDetails.userToken,
-          req.query.hasOwnProperty("createProject") ? gen.utils.convertStringToBoolean(req.query.createProject) : true
+          req.query.hasOwnProperty("createProject") ? gen.utils.convertStringToBoolean(req.query.createProject) : true,
+          req.query.createPrivateSolutionIfNotTargeted ? req.query.createPrivateSolutionIfNotTargeted : false
```

Comment:
@VISHNUDAS-tunerlabs - Let the app not tell whether to create a private solution or not. In importFromLibrary we can add check to not create a solution if its already created.